clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1368,7 +1374,14 @@ public void commitSync() {
      *
      * @throws org.apache.kafka.clients.consumer.CommitFailedException if the commit failed and cannot be retried.
      *             This can only occur if you are using automatic group management with {@link #subscribe(Collection)},
-     *             or if there is an active group with the same groupId which is using group management.
+     *             or if there is an active group with the same groupId which is using group management. In such cases,
+     *             when you are trying to commit to partitions that is no longer assigned to this consumer because the
+     *             consumer is for example no longer part of the group this exception would be thrown.
```

Comment:
If you try to commit and it fails because you no longer own some (or any) partitions because a rebalance just occurred, that's retriable. If you dropped out of the group and for that reason do not own those (or any) partitions, that's not retriable.
@guozhangwang I do agree that we could clarify the wording a bit: in particular, it seems to matter explicitly _why_ the consumer no longer owns those partitions